onebusaway-gtfs/src/main/java/org/onebusaway/gtfs/model/BookingRule.java

```diff
@@ -0,0 +1,311 @@
+/**
+ * Copyright (C) 2020 Victor Pavlushkov <victor@kyyti.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.onebusaway.gtfs.model;
+
+import org.onebusaway.csv_entities.schema.annotations.CsvField;
+import org.onebusaway.csv_entities.schema.annotations.CsvFields;
+
+@CsvFields(filename = "booking_rules.txt", required = false)
+public final class BookingRule extends IdentityBean<AgencyAndId> {
+
+  public static final int MISSING_VALUE = -999;
+
+  @CsvField(name = "booking_rule_id", mapping = EntityFieldMappingFactory.class)
+  private String id;
+
+  /** booking_type
+   (Enum, Required) Defines how much in advance the booking can be made. Value are:
+   0: Real-time booking only
+   1: Up to same-day booking, with advance notice
+   2: Up to prior day(s) booking
+   */
+  @CsvField(name = "booking_type", mapping = EntityFieldMappingFactory.class)
+  private int type;
+
+  /**
+   prior_notice_duration_min
+   (Integer, Conditionally Required) Minimum number of minutes of advance time necessary before travel to make a booking request.
+
+   Conditionally Required:
+   (The timing must be provided by either a duration or a last time & day)
+   Required for up-to-same-day booking (booking_type=1)..
+   Forbidden otherwise.
+   */
+  @CsvField(name = "prior_notice_duration_min", optional = true, mapping = EntityFieldMappingFactory.class)
```

Comment:
No need to declare the name, the default is derived from the field name, also no mapping needed